docs/ui_description.rst

```diff
@@ -0,0 +1,216 @@
+Glupyter Framework Overview
+===========================
+
+The glue-jupyter ("glupyter") package supports interacting with and
+visualizing data within the Jupyter environment using core elements from
+the Glue python package. It is distinct because unlike the more
+prominent Glue package, glupyter does not leverage Qt as the front-end
+gui library. Instead, glupyter maintains the separation between the data
+model (e.g. the core elements from Glue that are not dependent on the
+front-end library), and the view of the data (in this case, web-based
+tooling provided by Jupyter).
+
+Glupyter implements a base ``JupyterApplication`` object through which
+users can manage their data, create viewers, and add links between data
+sets. The data management and linking are controlled separate from the
+viewers in that changes made directly to the data state propagate to the
+UI (i.e. the UI acts strictly as the view to the data state, not the
+controller of the data state). The viewers themselves are based on the
+IPyWidget package which allows the creation of widgets that can be used
+and interacted with in python, but rendered in a javascript environment.
+
+There are two distinct use cases for the glupyter environment: first, as
+a means to procedurally interact with pieces of a user's workflow *in
+addition* to their work in e.g. a Jupyter Notebook; and second, to
+provide users a web-based gui to interact with and visualize their data
+while hiding the Jupyter context, e.g. a standalone web application.
+These two use cases describe a data-first and gui-first approach,
+respectively. This document will focus on detailing the design of the
+gui-first approach.
+
+General user interface design
+-----------------------------
+
+The general user interface is a parallel design to the Qt desktop
+interface. It is meant to be the standard scaffolding supporting the
+implementation of the individual viewers, data management functions, and
```

Comment:
implementation and display of?

To me, the general UI doesn't really support implementation, it's more about putting the pieces together?